Lib/test/test_subprocess.py

```diff
@@ -1360,6 +1360,27 @@ def test_communicate_epipe(self):
         self.addCleanup(p.stdin.close)
         p.communicate(b"x" * 2**20)
 
+    def test_string_representation(self):
+        """Test stirng representation Popen object without returncode."""
+        code = 'import sys; sys.exit(1)'
+        cmd = [sys.executable, '-c', code]
+        with subprocess.Popen(cmd) as proc:
+            args = ('%s...' % ' '.join(cmd)[:30]).strip()
+            result = "<subprocess.Popen: pid:'%d' args:'%s'>" % (
+                proc.pid, args)
+            self.assertEqual(proc.__repr__(), result)
+
+    def test_string_representation_with_retcode(self):
+        """Test stirng representation Popen object with returncode."""
+        code = 'import sys; sys.exit(1)'
```

Comment:
I'd suggest an error code that isn't 1, because 1 is common and we want to be sure we're getting the same error code. Choose some obscure error code like 57.